### ISSUE TITLE
Arcane Charges not displaying in Sample Sequence Table during 1st burn phase

### ISSUE THREAD

**Issue by SnowMewPlays** [simulationcraft/simc#3583]
Please include a brief overview of the issue in the title above, as well as completing the following:

#### Simulationcraft Version
simc-715-02-win64-a9d6335

#### If Simulationcraft is crashing, please include your dxdiag file - Instructions here - https://us.battle.net/support/en/article/Obtaining-System-Files
No crash

#### simc_gui.simc file or contents
This file is saved in the same folder as the program after each simulation. Make sure to run the simulation that is giving you issues, then include the file here by either uploading to github, copy/pasting the contents, or using http://pastebin.com/

#### Reproduction Steps:
1. Import an arcane mage (I'm using Snowkoa on Wyrmrest Accord on US servers)
2. Navigate to options, set up simulation Fight Style with LightMovement
2. Run simulation
3. In results, navigate to Action Priority List > Sample Sequence Table
4. Observe lack of arcane_charge buff during first part of the fight
5. Observe sudden appearance of arcane_charge following 1st movement phase OR after 1st occurrence  of arcane_barrage

Expected:
arcane_charge buff is built irl at the very start of the fight. This buff should persist throughout the whole simulation to confirm burn phase is staying at 4 - or to observe if barrage is being used on different fight types. And then to what charge the sim builds to following that situation.

Please include any other steps necessary to recreate the problem, or include logs/data to show what is occurring in game if there is an issue with how something is modeled.

**Comment by SnowMewPlays** [comment]
[simc_gui.zip](https://github.com/simulationcraft/simc/files/797836/simc_gui.zip)

**Comment by scamille** [comment]
Ok so this is just a reporting issue in the sample sequence table. Running simulation with debug=1 shows correct arcane_charge gains.

A bit more in-depth info for us developers:
The problem is that the buff has 'constant' set to true, and is thus not added to the sample sequence entries at first.
The buff has 'constant' true because the logic looks a bit too simple. If the buff is started at t=0, it is set to constant, and only later when it first expires constant is set to false.

So basically the Sample sequence code assumes a in-combat correctness of buff->constant, which the buff code just does not currently provide.

**Comment by seanpeters86** [comment]
Closing due to inactivity, feel free to reopen if you need more assistance.
